docs/guides/Package-authoring-best-practices.md

```diff
@@ -0,0 +1,177 @@
+| `Tags`                      | `PackageTags`                | `tags`                      | A space-delimited list of tags and keywords that describe the package. Tags are used when searching for packages. |
+| `Release notes`             | `PackageReleaseNotes`        | `releaseNotes`              | A description of the changes made in this release of the package.                                                 |
+
+### Package ID
+
+If you're publishing a completely new package:
+
+✔️ DO choose a package ID that is unique and clearly differentiated from existing packages on NuGet.org.
+> You can make sure a package ID is unique and differntiable by searching for the ID on NuGet.org or checking if the following link exists: https://www.nuget.org/packages/<package name\>.
+
+✔️ CONSIDER choosing a NuGet package name with a prefix that meets NuGet's prefix reservation [criteria](https://docs.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation#id-prefix-reservation-criteria).
+> Reserving the prefix ID for your package will let you get the verified check mark:
+> ![image](media/Verified-check-mark.png)
+> Check out the [Package ID prefix reservation docs](https://docs.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation) to learn more.
+
+### Package Version
+
+✔️ CONSIDER using [SemVer 2.0.0](https://semver.org/) to version your NuGet package.
+> Essentially, this means using the Major.Minor.Patch[-prerelease] format.
+
+✔️ DO include a pre-release suffix when releasing a non-stable package.
+
+See the [.NET library versioning guide](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/versioning) for more advanced guidance including aseembly versioning.
+
+### Authors
+
+✔️ CONSIDER including the current NuGet.org owners for consistency across NuGet clients.
+> Visual Studio currently displayed `Authors` on the Browse tab. It is also in the criteria for ID prefix reservation.
+
+### Description
+
+✔️ DO include a short description for your package that describes what it's for.
+> Package descriptions are one of the most prominent fields surfaced in NuGet search and will likely be the first thing a potential consumers looks at to determine if a package is right for them.
+
+### Copyright
+
+✔️ CONSIDER copyrighting your package with "Copyright (c) <name/company\> <year\>."
+>A copyright notice essentially indicates that your work cannot be copied without your permission. Including, a copyright notice in your package is easy and won't do any harm!
+
+Example: Copyright (c) Contoso 2020
+
+### Licensing
+
+✔️ DO specify a valid license for your package.
+> [!IMPORTANT]
+> A project without a license defaults to [exclusive copyright](https://choosealicense.com/no-permission/), making it legally impossible for other people to use.
+
+✔️ CONSIDER using a license expression (SPDX identifier).
```

Comment:
Would it be bad to use "Avoid" for license files? In general the best experience would be when users SPDX instead of taking the equivalent SPDX license text and putting it in a file. I think we should encourage people down that path as strongly as possible. SPDX allows for futures of filtering, grouping, etc. A license file is not very machine readable (although there is software that can do it with varying success).

Reply:
Personally, I'm wary of telling people to Avoid the proper use of a feature we continue to clearly support. I also think it may be redundant with with the suggestion to use license expression when possible.

In general, it's already more work for the user to include a license file over using a license expression. But in scenarios where a company is very legally cautious (license expression doesn't support signed licenses that are specific to the company/package) or where they have a custom license, I don't want to imply that having a license file is a "bad" practice.

Thoughts?

Reply:
Sounds good to me.